tests/test_backend.py

```diff
@@ -159,112 +159,124 @@ def test_get_all_metadata_simple(self, multi_backend_connection, backend1, backe
         assert metadata == [{"id": "S2"}, {"id": "S3"}]
 
     def test_get_all_metadata_common_collections_minimal(
-            self, multi_backend_connection, backend1, backend2, requests_mock
+            self, multi_backend_connection, backend1, backend2, requests_mock, flask_app
     ):
-        requests_mock.get(backend1 + "/collections", json={"collections": [{"id": "S3"}, {"id": "S4"}]})
-        requests_mock.get(backend2 + "/collections", json={"collections": [{"id": "S4"}, {"id": "S5"}]})
-        catalog = AggregatorCollectionCatalog(backends=multi_backend_connection)
-        metadata = catalog.get_all_metadata()
-        assert metadata == [
-            {"id": "S3"},
-            {
-                "id": "S4", "description": "S4", "title": "S4",
-                "stac_version": "0.9.0",
-                "extent": {"spatial": {"bbox": [[-180, -90, 180, 90]]}, "temporal": {"interval": [[None, None]]}},
-                "license": "proprietary",
-                "summaries": {"provider:backend": ["b1", "b2"]},
-                "links": [],
-            },
-            {"id": "S5"},
-        ]
+        with flask_app.app_context():
```

Comment:
just wondering: is this `with flask_app.app_context()` necessary?
it's not used in other tests

Reply:
I would also only put the necessary parts inside the `with` block.
E.g. the `request_mock` stuff and the final `assert` can stay outside I think
That makes it more clear for what the `app_context()` is necessary
and it will make reviewing this PR easier :)

Reply:
Since adding the url_for in the _merge_collection_metadata method it is required in all tests that go through this path. The url_for function is used to fill in the self, parent, and root links.

Reply:
I was afraid so. Then I think we should do the `app_context()`  call from the `flask_app` fixture instead of polluting each test with that implementation detail

Reply:
Done.